building.md

```diff
@@ -0,0 +1,86 @@
+# Build Vita3K
+
+## Windows (Visual Studio)
+
+- Install Visual Studio 2017 or 2019 and choose to install `Desktop development with C++` (as shown in this [image](https://i.imgur.com/bkY15Oh.png)). You will get compiler and `cmake` required for building.
+
+- Install `git` to `clone` the project. Download and install `git` from [here](https://git-scm.com).
+
+- Clone this repo.
+
+```cmd
+git clone --recursive https://github.com/Vita3K/Vita3K
+cd Vita3K
+```
+
+- Run `gen-windows.bat` to create Visual Studio project in `build-windows` directory.
+
+```cmd
+.\gen-windows
+```
+
+- Open the project generated in the `build-windows` directory.
+
+- Build and run it.
+
+## macOS (Xcode)
+
+- Install Xcode at App Store.
+
+- Install [`brew`](https://brew.sh).
+
+- Install dependencies with `brew`.
+
+```sh
+brew install git cmake boost
+```
+
+- Clone this repo.
+
+```sh
+git clone --recursive https://github.com/Vita3K/Vita3K
+cd Vita3K
+```
+
+- Generate Xcode project.
+
+```
+./gen-macos.sh
+```
+
+- Open Xcode project `Vita3K.xcodeproj` generated in `build-macos` directory.
+
+- When prompted to create schemes, create one for the `emulator` target only. The project builds many targets, so it will make your life easier if you create schemes as needed.
+
+- Build.
+
+### Linux (Debian)
+
+- Install dependencies.
+
+```sh
+sudo apt install git cmake libboost-all-dev libsdl2-dev
```

Comment:
I think this should be more than just Debian, as it's possible to build this on other distros. Instead of giving a debian command-line example, maybe just list the dependencies like: cmake, boost, sdl2, gtk3, pkg-config. (you missed the last 2)

Reply:
It is because we give instruction for installing dependencies with `apt`. For other package managers, if anyone is more familiar with it, we should let them put in the instructions. Yeah, I think we should have sub-sections for `Linux`.

About dependencies, I will re-check them and update. Thanks!